src/App/App.tsx

```diff
@@ -71,14 +74,47 @@ library.add(faExternalLinkSquareAlt);
 toast.configure({
   autoClose: TOAST_AUTO_CLOSE_DELAY /** defines how long a toast remains visible on screen */,
 });
-initGoogleAnalytics();
+
+const initializeOptions = {
+  testMode: GA_ENV_TEST === GA_ENV,
+};
+
+/** Initialize google analytics */
+initGoogleAnalytics(GA_CODE, initializeOptions);
 
 import store from '../store';
 import { getOauthProviderState } from '../store/selectors';
 import './App.css';
 
+/** Interface defining component state */
+export interface AppState {
+  username?: string;
+}
+
 /** Main App component */
-class App extends Component {
+class App extends Component<{}, AppState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      username: (getUser(store.getState()) || {}).username || '',
+    };
+  }
+
+  public componentDidUpdate(_: {}, prevState: AppState) {
+    const username = (getUser(store.getState()) || {}).username || '';
+
+    if (prevState.username !== username) {
```

Comment:
should we check that the GA code is provided here? If it's not then we don't need to worry about the GA dimensions or who the user is (at this level).